From 99895c8daf6073fa41404acf324dc6334f235b79 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 14 Jul 2014 13:41:34 -0400
Subject: [x86] mce: Update MCE severity condition check

Message-id: <1405345303-24468-5-git-send-email-prarit@redhat.com>
Patchwork-id: 85780
O-Subject: [RHEL7.1 PATCH BZ 1119306 04/13] x86/mce: Update MCE severity condition check
Bugzilla: 1119306
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1119306

commit 33d7885b594e169256daef652e8d3527b2298e75
Author: Chen Gong <gong.chen@linux.intel.com>
Date:   Thu Jun 20 05:16:12 2013 -0400

    x86/mce: Update MCE severity condition check

    Update some SRAR severity conditions check to make it clearer,
    according to latest Intel SDM Vol 3(June 2013), table 15-20.

    Signed-off-by: Chen Gong <gong.chen@linux.intel.com>
    Acked-by: Naveen N. Rao <naveen.n.rao@linux.vnet.ibm.com>
    Signed-off-by: Tony Luck <tony.luck@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/cpu/mcheck/mce-severity.c b/arch/x86/kernel/cpu/mcheck/mce-severity.c
index beb1f16..e270352 100644
--- a/arch/x86/kernel/cpu/mcheck/mce-severity.c
+++ b/arch/x86/kernel/cpu/mcheck/mce-severity.c
@@ -110,22 +110,17 @@ static struct severity {
  /* known AR MCACODs: */
 #ifdef CONFIG_MEMORY_FAILURE
  MCESEV(
-  KEEP, "HT thread notices Action required: data load error",
-  SER, MASK(MCI_STATUS_OVER|MCI_UC_SAR|MCI_ADDR|MCACOD, MCI_UC_SAR|MCI_ADDR|MCACOD_DATA),
-  MCGMASK(MCG_STATUS_EIPV, 0)
+  KEEP, "Action required but unaffected thread is continuable",
+  SER, MASK(MCI_STATUS_OVER|MCI_UC_SAR|MCI_ADDR|MCACOD, MCI_UC_SAR|MCI_ADDR),
+  MCGMASK(MCG_STATUS_RIPV, MCG_STATUS_RIPV)
   ),
  MCESEV(
-  AR, "Action required: data load error",
+  AR, "Action required: data load error in a user process",
   SER, MASK(MCI_STATUS_OVER|MCI_UC_SAR|MCI_ADDR|MCACOD, MCI_UC_SAR|MCI_ADDR|MCACOD_DATA),
   USER
   ),
  MCESEV(
-  KEEP, "HT thread notices Action required: instruction fetch error",
-  SER, MASK(MCI_STATUS_OVER|MCI_UC_SAR|MCI_ADDR|MCACOD, MCI_UC_SAR|MCI_ADDR|MCACOD_INSTR),
-  MCGMASK(MCG_STATUS_EIPV, 0)
-  ),
- MCESEV(
-  AR, "Action required: instruction fetch error",
+  AR, "Action required: instruction fetch error in a user process",
   SER, MASK(MCI_STATUS_OVER|MCI_UC_SAR|MCI_ADDR|MCACOD, MCI_UC_SAR|MCI_ADDR|MCACOD_INSTR),
   USER
   ),
-- 
1.7.1